src/core_plugins/kibana/public/context/index.js

```diff
@@ -0,0 +1,56 @@
+import uiRoutes from 'ui/routes';
+import './app';
+import { getDocumentUid } from './api/utils/ids';
+
+
+uiRoutes
+.when('/context/:indexPattern/:type/:id', {
+  controller: ContextAppRouteController,
+  controllerAs: 'contextAppRoute',
+  resolve: {
+    indexPattern: function ($route, courier, savedSearches) {
+      return courier.indexPatterns.get($route.current.params.indexPattern);
+    },
+  },
+  template: (
+    `<context-app
+      anchor-uid="contextAppRoute.anchorUid"
+      columns="contextAppRoute.state.columns"
+      discover-url="contextAppRoute.discoverUrl"
+      index-pattern="contextAppRoute.indexPattern"
+      predecessor-count="contextAppRoute.state.predecessorCount"
+      successor-count="contextAppRoute.state.successorCount"
+      sort="[contextAppRoute.indexPattern.timeFieldName, 'desc']"
+    >`
+  ),
+});
+
+
+function ContextAppRouteController(
```

Comment:
So if we put this controller in a separate file, it should be grouped in a directory with the route definition and the template. I really like the approach that has been taken in the watcher ui, but a subdirectory in `sections` feels like overkill for this single-route app. The `context` directory itself is the section directory. That's why I put its route definition in `index.js`. I'm struggling to find a place to put this very route-specific controller if not in the same file.